### PR TITLE
Introduce ModelPlotConfig.annotations_enabled setting

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR introduces user-visible setting (`ModelPlotConfig.annotations_enabled`) for enabling/disabling model change annotations.
The new setting is placed in `ModelPlotConfig` as the model plots and annotations show up together in the UI. However, the user is able to enable/disable model change annotations (`ModelPlotConfig.annotations_enabled`) independently of enabling/disabling model plot itself (`ModelPlotConfig.enabled`).

Relates https://github.com/elastic/elasticsearch/issues/55781

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/config/ModelPlotConfigTests.java [hunk 1]
```diff
@@ -25,7 +25,11 @@
 
     @Override
     protected ModelPlotConfig createTestInstance() {
-        return new ModelPlotConfig(randomBoolean(), randomAlphaOfLengthBetween(1, 30));
+        return createRandomized();
+    }
+
+    public static ModelPlotConfig createRandomized() {
+        return new ModelPlotConfig(randomBoolean(), randomAlphaOfLengthBetween(1, 30), randomBoolean());
```
Comment (benwtrent) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/config/ModelPlotConfigTests.java:32]: Would be good to have a `null` case here for `annotationsEnabled`

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfigTests.java [hunk 2]
```diff
@@ -15,13 +15,19 @@
...
     }
 
     @Override
     protected ModelPlotConfig createTestInstance() {
-        return new ModelPlotConfig(randomBoolean(), randomAlphaOfLengthBetween(1, 30));
+        return createRandomized();
+    }
+
+    public static ModelPlotConfig createRandomized() {
+        return new ModelPlotConfig(randomBoolean(), randomAlphaOfLengthBetween(1, 30), randomBoolean());
```
Comment (benwtrent) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfigTests.java:30]: would be good to test the `null` cases.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/config/ModelPlotConfigTests.java (hunk 1 above)
Comment (przemekwitek) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/config/ModelPlotConfigTests.java]: Done.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfigTests.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfigTests.java]: Done.

**Review by benwtrent — APPROVED** [review]

**Pull request merged**